Dockerfile

```diff
@@ -1,28 +1,54 @@
-FROM python:3.12-slim
+ARG BASE_IMAGE=python:3.12-slim
+FROM $BASE_IMAGE AS base
 
+ARG REQUIREMENTS=requirements_production.txt
+
+# Set environment variables
+ENV FLASK_APP=govuk-frontend-flask.py
 ENV FLASK_RUN_HOST=0.0.0.0
 ENV FLASK_RUN_PORT=8000
+ENV PYTHONUNBUFFERED=1
 
-# Set the working directory in the container
-WORKDIR /usr/src/app
+RUN apt-get install tzdata
 
-# Change ownership of the working directory to the non-root user
-RUN chown -R 1000:1000 /usr/src/app
+# Create a non-root user
+RUN adduser --disabled-password app && \
+    cp /usr/share/zoneinfo/Europe/London /etc/localtime
```

Comment:
Updated to remove tzdata, can you provide further clarity on the second part of the comment?

Reply:
The second part you can ignore if we're not installing tzdata as it's about setting it up while it installs, because apparently when you install it stops to ask you the timezone. 

https://askubuntu.com/questions/909277/avoiding-user-interaction-with-tzdata-when-installing-certbot-in-a-docker-contai